examples/platforms/simulation/radio.c

```diff
@@ -436,6 +478,9 @@ otError otPlatRadioSleep(otInstance *aInstance)
     {
         error  = OT_ERROR_NONE;
         sState = OT_RADIO_STATE_SLEEP;
+#if OPENTHREAD_SIMULATION_VIRTUAL_TIME == 1
+        reportRadioStatusToOTNS();
```

Comment:
Should we add a method for radio state transferring and add reportRadioStatusToOTNS to that method?

Reply:
HI @simonlingoogle 
Thanks for the review. I could not fully understand your question here. However, if it helps, in line 225-258 of Radio.c, we have this function declared when "OTNS=1".

https://github.com/Vinggui/openthread/blob/2b38d970715d0dec2739eae0458465ba68abb90f/examples/platforms/simulation/radio.c#L225-L258

 If I did not help with your question, please make it again, so that I can better satisfy this point. Thanks!

Reply:
What I mean is that instead add callings to reportRadioStatusToOTNS(); at multiple places in code, we can define a new function for changing radio state and call reportRadioStatusToOTNS() in that method.
e.g.

```c++
void setRadioState(otRadioState aState)
{
    sState = aState;
#if OPENTHREAD_SIMULATION_VIRTUAL_TIME == 1
    reportRadioStatusToOTNS();
#endif
}
```

Then, we can change `sState = ...` assignments to setRadioState(...);

Thoughts?

Reply:
Oh, I see. I like your idea, makes total sense since now we have multiple similar calls around the code, this would make it better.
Thank you, and check the next commit coming out.